packages/backend-for-frontend/src/services/catalogService.ts

```diff
@@ -1786,5 +1786,27 @@ export function catalogServiceBuilder(
 
       return { id: descriptorId };
     },
+    async checkEServiceNameAvailability(
+      name: string,
+      { headers, logger, authData }: WithLogger<BffAppContext>
+    ): Promise<boolean> {
+      logger.info(
+        `Checking e-service name availability ${name} for producer ${authData.organizationId}`
+      );
+
+      const templateInstances = await getAllFromPaginated((offset, limit) =>
+        catalogProcessClient.getEServices({
+          headers,
+          queries: {
+            limit,
+            offset,
+            producersIds: [authData.organizationId],
+            name,
+          },
+        })
+      );
```

Comment:
given a name and a producer, only 0 or 1 eservices can exist
```suggestion
      const templateInstances =
        await catalogProcessClient.getEServices({
          headers,
          queries: {
            limit: 1,
            offset: 0,
            producersIds: [authData.organizationId],
            name,
          },
        });
```

Reply:
Kinda the same reason as stated [here](https://github.com/pagopa/interop-be-monorepo/pull/1680#discussion_r2005944837).
In this case we would depend on the order the result arrives, and it seems fragile to me